lib/src/backend.rs

```diff
@@ -458,7 +444,8 @@ pub trait Backend: Send + Sync + Debug {
         sign_with: Option<&mut SigningFn>,
     ) -> BackendResult<(CommitId, Commit)>;
 
-    /// Get all copy records for `paths` in the dag range `roots..heads`.
+    /// Get copy records for the dag range `root..head`.  If `paths` is empty
+    /// include all paths, otherwise restrict to only `paths`.
```

Comment:
nit: use `Option<&[RepoPathBuf]>`? It can be source of bugs if `[]` had a different meaning.

Reply:
done